src/components/Input/Button.js

```diff
@@ -102,35 +98,38 @@ const StyledButton = styled.a`
     `color: ${p.theme.colors.background};
     background: ${p.theme.colors.primary};
     :hover {
-      ${
-        p.disabled
-          ? `
-      cursor: not-allowed;
-      `
-          : `
-      background: ${p.theme.colors.tertiaryHover};
-      `
-      }
+      ${p.disabled ? `cursor: not-allowed;` : `background: ${p.theme.colors.tertiaryHover};`}
     }
     :focus {
       box-shadow: 0 0 0 .2rem ${hexToRgba(p.theme.colors.primary, 0.5)};
     }`}
     ${p =>
+    p.color === 'warning' && // warning color (solid button)
+    `color: ${p.theme.colors.warning};
+    border: 1px solid ${p.theme.colors.warning};
+      :hover {
+        ${
+          p.disabled
+            ? `cursor: not-allowed;`
+            : `background: ${hexToRgba(p.theme.colors.secondaryWarning, 0.2)};`
+        }
+      }
+      :focus {
+        box-shadow: 0 0 0 .2rem ${hexToRgba(p.theme.colors.warning, 0.5)};
+      }`}
+    ${p =>
```

Comment:
thoughts on pulling this out into a `const WarningButton = styled(StyledButton)...`? this component is getting really large

Reply:
I agree it's getting super big idk if making a new component is the best thing? Since the changes are so similar to the changes made by primary/secondary/tertiary but idk what else we can do

Reply:
do you think we can switch on p.color and simplify the if statements